src/games/strategy/engine/data/GameParser.java

```diff
@@ -466,22 +467,9 @@ private Object getInstance(final String className) throws GameParseException {
    */
   private Class<?> getClassByName(final String className) throws GameParseException {
     try {
-      final Class<?> instanceClass = Class.forName(className);
-      return instanceClass;
-    }
-    // if class cannot be found than it is either not a valid class or an old class that was deleted/renamed
-    catch (final ClassNotFoundException cnfe) {
-      if (newClassesForOldNames == null) {
-        newClassesForOldNames = new HashMap<>();
-        // put in here class names that have been changed like //newClassesForOldNames.put("<oldClassName>",
-        // "<newClassName>"), e.g.
-        // newClassesForOldNames.put("attachment", "attachment")
-      }
-      final String newClassName = newClassesForOldNames.get(className);
-      if (newClassName != null) {
-        return getClassByName(newClassName);
-      }
-      throw new GameParseException(mapName, "Class <" + className + "> could not be found.");
+      return Class.forName(className);
+    } catch (final ClassNotFoundException e) {
+      throw new GameParseException(mapName, "Class <" + className + "> could not be found: " + e);
```

Comment:
GameParseException does not have any constructors to accept the `ClassCastException`. Beyond that, I'd rather not add it for this purpose. Knowing the name of the class not found and anything from the exception message should be more than enough to know what the problem is (the line number in the XML would be great, but stack trace will not give that). For that matter, a stack trace is liable to be more clutter than anything else, knowing the entry point to this exception is not very helpful in fixing the problem.